### PR TITLE
Change await log type to cloud-ready-check lib

### PR DESCRIPTION
**Pull request by lblackstone**
### Proposed changes
The cloud-ready-check library includes an identical
logging.Message type definition, so remove the type
from the provider and update all references to use the
external type instead.

### CODE REVIEW

File: provider/pkg/logging/time_ordered_log_set.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2016-2019, Pulumi Corporation.
+// Copyright 2016-2022, Pulumi Corporation.
```
Comment (viveklak) [inline provider/pkg/logging/time_ordered_log_set.go:1]: Should this file be in cloud-ready-checks instead?

File: provider/pkg/logging/time_ordered_log_set_test.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2016-2019, Pulumi Corporation.
```
Comment (viveklak) [inline provider/pkg/logging/time_ordered_log_set_test.go:1]: Same here

File: provider/pkg/logging/dedup_logger.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2016-2019, Pulumi Corporation.
```
Comment (viveklak) [inline provider/pkg/logging/dedup_logger.go:1]: Same here.

File: provider/pkg/logging/time_ordered_log_set.go (hunk 1 above)
Comment (lblackstone) [reply provider/pkg/logging/time_ordered_log_set.go:1]: I think this logic is more specific to provider logging, so I decided to leave it here. The await library returns the messages and it's up to the client how to handle them.

**Review by viveklak — APPROVED** [review]

**Pull request merged**